validation/pipeline/tfp/Jenkinsfile.airgap.tests

```diff
@@ -0,0 +1,180 @@
+                                    $class: 'GitSCM',
+                                    branches: [[name: "*/${testBranch}"]],
+                                    extensions: scm.extensions + [[$class: 'CleanCheckout']],
+                                    userRemoteConfigs: testRepo
+                                ])
+                    }
+
+                    dir ("./") {
+                        stage('Configure and Build') {
+                          if (env.AWS_SSH_PEM_KEY && env.AWS_SSH_KEY_NAME) {
+                            dir("./validation/.ssh") {
+                              def decoded = new String(AWS_SSH_PEM_KEY.decodeBase64())
+                              writeFile file: AWS_SSH_KEY_NAME, text: decoded
+                            }
+                          }
+                          dir("./validation") {                            
+                            config = config.replace('${PRIVATE_REGISTRY}', privateRegistry)
+                            config = config.replace('${RANCHER_ADMIN_TOKEN}', adminToken)
+            
+                            writeFile file: filename, text: config
+                            env.CATTLE_TEST_CONFIG = testRootPath+filename
+                          }
+                          dir ("./") {
+                            sh "./validation/configure.sh"
+                            sh "docker build . -f ./validation/Dockerfile.validation --build-arg CORRAL_VERSION=${env.CORRAL_VERSION} -t ${imageName}"
+                            sh "docker volume create --name ${validationVolume}"
+                          }
+                        }
+                        stage('Run Validation Tests') {
+                          try {
+                            sh """
+                            docker run -v ${validationVolume}:/root --name ${golangTestContainer} -t --env-file ${envFile} ${imageName} sh -c "
+                            gotestsum --format standard-verbose --packages=\"${golangTestDir}\" --junitfile ${testResultsOut} -- -tags=\"airgap\" -timeout=${timeout}"
+                            """
+
+                              
+                          } catch(err) {
+                            echo 'Validation tests had failures. Aborting'
+                          }
+                          sh """
+                          docker rm ${golangTestContainer}
+                          docker volume rm ${validationVolume}
+                          docker rmi ${imageName}
+                          
+                          """
+                        }
+            
+                        stage('Cleanup terraform resources'){
+                            sh """
+                            docker run --rm -v ./root/go/src/github.com/rancher/tfp-automation/modules/airgap:/terraform-files \
+                                -w /terraform-files hashicorp/terraform:latest \
+                                terraform init && terraform destroy -auto-approve
```

Comment:
nit; I feel this could be more generic but this can be done/addressed in future PR